src/js/core/methods/CardanoSignTransaction.js

```diff
@@ -155,38 +207,136 @@ export default class CardanoSignTransaction extends AbstractMethod {
     _ensureFirmwareSupportsParams() {
         const { params } = this;
 
-        params.certificates.forEach(certificate => {
+        params.certificatesWithPoolOwnersAndRelays.forEach(({ certificate }) => {
             if (certificate.type === CardanoCertificateType.STAKE_POOL_REGISTRATION) {
                 this._ensureFeatureIsSupported('SignStakePoolRegistrationAsOwner');
             }
         });
 
-        if (params.validity_interval_start != null) {
+        if (params.validityIntervalStart != null) {
             this._ensureFeatureIsSupported('ValidityIntervalStart');
         }
 
-        params.outputs.forEach(output => {
-            if (output.token_bundle && output.token_bundle.length > 0) {
+        params.outputsWithTokens.forEach(output => {
+            if (output.tokenBundle && output.tokenBundle.length > 0) {
                 this._ensureFeatureIsSupported('MultiassetOutputs');
             }
         });
 
-        if (params.auxiliary_data) {
+        if (params.auxiliaryData) {
             this._ensureFeatureIsSupported('AuxiliaryData');
         }
     }
 
-    async run() {
-        this._ensureFirmwareSupportsParams();
+    async _sign_tx_streamed(): Promise<CardanoSignedTxData> {
+        const typedCall = this.device.getCommands().typedCall.bind(this.device.getCommands());
 
+        const hasAuxiliaryData = !!this.params.auxiliaryData;
+
+        let signTxInitMessage = {
+            signing_mode: this.params.signingMode,
+            protocol_magic: this.params.protocolMagic,
+            network_id: this.params.networkId,
+            inputs_count: this.params.inputsWithPath.length,
+            outputs_count: this.params.outputsWithTokens.length,
+            fee: this.params.fee,
+            ttl: this.params.ttl,
+            certificates_count: this.params.certificatesWithPoolOwnersAndRelays.length,
+            withdrawals_count: this.params.withdrawals.length,
+            has_auxiliary_data: hasAuxiliaryData,
+            validity_interval_start: this.params.validityIntervalStart,
+            witnesses_count: this.params.witnessPaths.length,
+        };
+
+        // init
+        await typedCall('CardanoSignTxInit', 'CardanoTxItemAck', signTxInitMessage);
```

Comment:
I wasn't able to think of a nicer way to send the messages one by one, but perhaps it can be done nicer?